cmd/podman/pull.go

```diff
@@ -124,7 +124,12 @@ func pullCmd(c *cli.Context) error {
 		}
 		imgID = newImage[0].ID()
 	} else {
-		newImage, err := runtime.ImageRuntime().New(getContext(), image, c.String("signature-policy"), c.String("authfile"), writer, &dockerRegistryOptions, image2.SigningOptions{}, true, forceSecure)
+		authfile := c.String("authfile")
+		envAuthfile := os.Getenv("REGISTRY_AUTH_FILE")
+		if authfile == "" {
+			authfile = envAuthfile
+		}
+		newImage, err := runtime.ImageRuntime().New(getContext(), image, c.String("signature-policy"), authfile, writer, &dockerRegistryOptions, image2.SigningOptions{}, true, forceSecure)
```

Comment:
Sorry, just thought of this now.  Since we'll need to do this code in a number of other calls, it might be good to create a function in cmd/podman/utils.go to do this processing.  Then call something like `utils.GetAuthFile(c.String("authfile"))` here which would return a string with the authfile location or "" if not specified.